src/Middleware/Rewrite/src/ApacheModRewrite/ConditionEvaluator.cs

```diff
@@ -42,14 +42,14 @@ public static MatchResults Evaluate(IEnumerable<Condition> conditions, RewriteCo
 
                 if (condResult.Success && trackAllCaptures && prevBackReferences != null)
                 {
-                    prevBackReferences.Add(currentBackReferences);
+                    prevBackReferences.Add(currentBackReferences!);
                     currentBackReferences = prevBackReferences;
                 }
 
                 prevBackReferences = currentBackReferences;
             }
 
-            return new MatchResults { BackReferences = prevBackReferences, Success = condResult.Success };
+            return new MatchResults(condResult!.Success, prevBackReferences);
```

Comment:
💡 Consider proper argument validation instead of suppression. If `conditions` is empty when this method is called, a `NullReferenceException` will be thrown here.

Reply:
I looked at the source and this method will only be called if there are conditions, which means there will be a condResult. I've added a Debug.Assert with message to make that explicit.